wireguard@nicoulaj.net/files/wireguard@nicoulaj.net/applet.js

```diff
@@ -0,0 +1,212 @@
+const Applet = imports.ui.applet;
+const PopupMenu = imports.ui.popupMenu;
+const GLib = imports.gi.GLib;
+const Gio = imports.gi.Gio;
+const Gettext = imports.gettext;
+const Lang = imports.lang;
+const ModalDialog = imports.ui.modalDialog;
+
+const UUID = "wireguard@nicoulaj.net";
+
+Gettext.bindtextdomain(UUID, GLib.get_home_dir() + "./local/share/locale");
+
+function _(str) {
+    return Gettext.dgettext(UUID, str)
+}
+
+function main(metadata, orientation) {
+    return new WireGuardApplet(orientation);
+}
+
+const WireGuardApplet = class WireGuardApplet extends Applet.IconApplet {
+
+    constructor(orientation, panel_height, instance_id) {
+        super(orientation, panel_height, instance_id);
+
+        this._orientation = orientation;
+        this._menu_manager = null;
+        this._menu = null;
+        this._net_monitor = null;
+        this._net_monitor_id = null;
+        this._net_interfaces = null;
+        this._wg_monitor = null;
+        this._wg_monitor_id = null;
+        this._wg_interfaces = null;
+
+        this.set_applet_icon_symbolic_name("off");
+        this.set_applet_tooltip(_("WireGuard"));
+    }
+
+    on_applet_added_to_panel(userEnabled) {
+
+        if (!this._net_monitor) {
+            this._net_monitor = Gio.network_monitor_get_default();
+            this._net_monitor_id = this._net_monitor.connect('network-changed', (monitor, network_available, user_data) => this._on_net_changed());
+        }
+
+        if (!this._wg_monitor) {
+            this._wg_monitor = Gio.file_new_for_path("/etc/wireguard").monitor_directory(Gio.FileMonitorFlags.SEND_MOVED, null);
```

Comment:
Might be smoother to check if wireguard is installed, something like this would work.

```js
let path = Gio.file_new_for_path("/etc/wireguard");
if (!path.query_exists(null)) {
  Main.notify('Please install wireguard')
  // ... notify the user and let the applet load, otherwise they may never know and the applet will remain enabled on cinnamon restart.
  return;
}
```

Reply:
thanks, updated